File: 7_kyu/Recursion_101.py
from typing import List

def solve(a: int, b: int) -> List[int]:
    while 1:
        if a == 0 or b == 0:
            return [a,b]
        a %= 2*b
        if b >= 2*a:
            b -= 2*a
        else:
            return [a,b]
    

File: 7_kyu/Recursion_101.go
package kata

func Solve(a, b int) []int {
	for a > 0 && b > 0 && (a >= 2*b || b >= 2*a) {
		if b > 0 {
			a %= 2 * b
		}
		if a > 0 {
			b %= 2 * a
		}
	}
	return []int{a, b}
}
